cg/models/orders/validators/excel_sample_validators.py

```diff
@@ -33,11 +33,19 @@ def validate_data_analysis(data_analysis):
 
 
 def numeric_value(value: Optional[str]):
+    """Validates that the given string can be given as either an integer or a float. Also converts floats of the
+    type x.00 to x."""
     if not value:
         return None
-    str_value = value.rsplit(".0")[0]
-    if str_value.replace(".", "").isnumeric():
-        return str_value
+    if value.replace(".", "").isnumeric():
+        try:
+            float_value = float(value)
+            if float_value.is_integer():
+                return value.split(".")[0]
+            return value
+        except ValueError:
+            if value.isnumeric():
+                return value
```

Comment:
This validator is confusing to me, do you think it can be made clearer somehow? Here are some suggestions, feel free to disregard:

```python
from typing import Optional

def numeric_value(value: Optional[str]) -> Optional[str]:
    """Validates that the given string can be cast to either an integer or a float.
    Also converts floats of the type x.00 to x."""
    
    if value is None or value == '':
        return None # I would expect this to raise a value error, but seems like the field is optional?
    
    try:
        float_value = float(value)
    except ValueError:
        return None  # Cannot convert to a float, return None
    
    if float_value.is_integer():
        return str(int(float_value))  # Return as integer string if it's an integer
    
    return str(float_value)  # Return as float string

```

Reply:
Not sure about catching the exceptions though and whether it makes sense to do that in a validator

Reply:
I think that looks much better! Might be worth raising the AttributeError if we can not parse the string as a float as to not change the behaviour, though.